packages/utils/src/lib/file-system.ts

```diff
@@ -85,6 +87,7 @@ export async function importEsmModule(options: Options): Promise<unknown> {
     throw new NoExportError(options.filepath);
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-member-access
   return mod.default;
```

Comment:
Weird that ESLint complains about this. We assign it to `unknown` return type, which I would consider very safe :sweat_smile:

Reply:
The problem is, `mod` is `any` as returned by `bundleRequire`
![image](https://github.com/code-pushup/cli/assets/6306671/732a2821-121a-429f-b569-287db1ce04fb)

One can type the function itself if they know what to expect in `mod`.

Reply:
> One can type the function itself if they know what to expect in mod. But when I do bundleRequire<unknown> then the (!('default' in mod)) complains 😅

You're on the right track :slightly_smiling_face: Just needs some tweak so that `in` operator is safe, TS needs to know `mod` is an object. Then no ESLint disables are necessary.

- If you cast it as `object`, the error goes away.
  ```ts
  export async function importEsmModule(options: Options): Promise<unknown> {
    const { mod } = await bundleRequire<object>({
      format: 'esm',
      ...options,
    });

    if (!('default' in mod)) {
      throw new NoExportError(options.filepath);
    }

    return mod.default;
  }
  ```
- If you cast it as `unknown`, then you need to add runtime checks to verify it's an object.
  ```ts
  export async function importEsmModule(options: Options): Promise<unknown> {
    const { mod } = await bundleRequire<unknown>({
      format: 'esm',
      ...options,
    });

    if (mod == null || typeof mod !== 'object' || !('default' in mod)) {
      throw new NoExportError(options.filepath);
    }

    return mod.default;
  }
  ```